ee_plugin/ui/forms/add_image_collection.py

```diff
@@ -0,0 +1,278 @@
+    percentile_value: Optional[int] = None,
+):
+    """
+    Loads and optionally filters an ImageCollection, then adds it to the map.
+
+    Args:
+        image_collection_id (str): The Earth Engine ImageCollection ID.
+        start_date (str, optional): Start date (YYYY-MM-DD) for filtering.
+        end_date (str, optional): End date (YYYY-MM-DD) for filtering.
+        extent (ee.Geometry, optional): Geometry to filter (or clip) the ImageCollection.
+        filters (list, optional): List of property filters to apply.
+
+    Returns:
+        ee.ImageCollection: The filtered ImageCollection.
+    """
+    ic = ee.ImageCollection(image_collection_id)
+
+    if filters:
+        # Define a dictionary to map operators to corresponding ee.Filter functions
+        filter_functions = {
+            "Equals (==)": ee.Filter.eq,
+            "Not Equals (!=)": ee.Filter.neq,
+            "Less Than (<)": ee.Filter.lt,
+            "Greater Than (>)": ee.Filter.gt,
+            "Less Than or Equal (<=)": ee.Filter.lte,
+            "Greater Than or Equal (>=)": ee.Filter.gte,
+        }
+
+        for f in filters:
+            filter_name, filter_operator, filter_value = f
+            # Convert filter_value to float if it's a valid number, otherwise keep as string
+            filter_value = (
+                float(filter_value)
+                if filter_value.replace(".", "", 1).isdigit()
+                else filter_value
+            )
+            filter_func = filter_functions.get(filter_operator)
+            if filter_func:
+                ic = ic.filter(filter_func(filter_name, filter_value))
+            else:
+                # Log an error if the filter operator is invalid
+                # somewhat hot take:
+                # do not add ImageCollection to the map
+                # if the filter operator is invalid, could lead to unexpected results for the user
+                logger.error(f"Invalid filter operator: {filter}.")
+                return
+
+    # Apply date filter if specified
+    if start_date and end_date:
+        ic = ic.filter(ee.Filter.date(ee.Date(start_date), ee.Date(end_date)))
+
+    # Apply spatial filter if extent is specified
+    if extent:
+        ic = ic.filterBounds(ee.Geometry.Rectangle(extent))
+
+    # Get visualization parameters from user as JSON
+    if not viz_params:
+        viz_params = {}
+    elif isinstance(viz_params, str):  # Ensure it's a string before replacing quotes
+        try:
+            viz_params = json.loads(viz_params.replace("'", '"'))
+        except json.JSONDecodeError as e:
+            logger.error(f"Invalid JSON format in visualization parameters: {str(e)}")
+            raise json.JSONDecodeError(
+                "Invalid JSON format in visualization parameters",
+                viz_params,  # The problematic string
+                0,  # Position of error (use actual error position if available)
+            )
```

Comment:
When recasting errors like this, it's best to instead [_chain_ errors](https://docs.python.org/3/tutorial/errors.html#exception-chaining):

```suggestion
            raise BadInput(
                "Invalid JSON format in visualization parameters",
                viz_params,  # The problematic string
                0,  # Position of error (use actual error position if available)
            ) from e
```

This way, we can coerce an error to something that we control (e.g. `BadInput`) and handle however we want, or let it raise and the stack trace will demonstrate what caused the error.